respy/simulate.py

```diff
@@ -138,25 +138,39 @@ def _simulate_data(state_space, base_draws_sim, base_draws_wage, optim_paras, op
 
     base_draws_wage_transformed = np.exp(base_draws_wage * optim_paras["meas_error"])
 
+    # Create observables for simulation and store them in an extra container that
+    # will be added to the state space container later
+    container_obs = ()
+    for observable in optim_paras["observables"].keys():
+        container_obs += (
+            _get_random_initial_observable(observable, options, optim_paras),
+        )
+
+    # Create a DataFrame to match columns to covariates. Is changed in-place.
+    if container_obs != ():
+        states_df = pd.DataFrame(
+            np.column_stack(container_obs),
+            columns=[x for x in optim_paras["observables"].keys()],
+        ).assign(period=0)
+    else:
+        states_df = pd.DataFrame().assign(period=0)
+
     # Create initial experiences, lagged choices and types for agents in simulation.
     container = ()
     for choice in optim_paras["choices_w_exp"]:
-        container += (_get_random_initial_experience(choice, optim_paras, options),)
-
-    # Create a DataFrame to match columns to covariates. Is changed in-place.
-    states_df = pd.DataFrame(
-        np.column_stack(container),
-        columns=[f"exp_{i}" for i in optim_paras["choices_w_exp"]],
-    ).assign(period=0)
+        states_df[f"exp_{choice}"] = _get_random_initial_experience(
+            choice, optim_paras, options
+        )
+        container += (states_df[f"exp_{choice}"],)
 
     for lag in reversed(range(1, n_lagged_choices + 1)):
         container += (_get_random_lagged_choices(states_df, optim_paras, options, lag),)
 
+    container += container_obs
     container += (_get_random_types(states_df, optim_paras, options),)
 
     # Create a matrix of initial states of simulated agents.
     current_states = np.column_stack(container).astype(np.uint8)
```

Comment:
Instead of the container, create a variable `state_space_cols` to select the state space dimensions in the correct order. Then, `current_states = states_df[state_space_cols].to_numpy(dtype=np.uint8)` instead of the container stacking.